src/main/java/teammates/storage/api/CommentsDb.java

```diff
@@ -406,12 +402,7 @@ public void deleteCommentsByStudentEmail(String courseId, String email) {
         Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, email);
 
         // student right now cannot be giver, so no need to&should not check for giver
-        List<Comment> recipientComments = this.getCommentEntitiesForRecipients(courseId,
-                                                       CommentParticipantType.PERSON, email);
-
-        getPm().deletePersistentAll(recipientComments);
-
-        getPm().flush();
+        getCommentsForRecipientsQuery(courseId, CommentParticipantType.PERSON, email).deletePersistentAll();
```

Comment:
The first step just constructs the data transfer object though - is that considered a step? I felt it's similar to `getPm()`.